### PR TITLE
Minimum number of challenge validators should be respected

### LINKED ISSUE #1711: Minimum number of challenge validators must respected
A challenge must be validated by `the number of data shards assigned to the allocation + 1` validators. If we don’t have enough validators, we should return an error instead of continuing. 

Add the verification in the next place.
https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/challenge.go#L881

### PR DESCRIPTION
**Pull request by devyetii**
## Fixes
#1711 
## Changes
- Check number of validators before proceeding with challenge creation and never complete creation if not enough
## Tests
Tasks to complete before merging PR:
- [x]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | 1711-minimum-number-of-challenge-validators-must-respected  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/smartcontract/storagesc/challenge.go [hunk 1]
```diff
@@ -970,12 +970,30 @@ func (sc *StorageSmartContract) populateGenerateChallenge(
...
 	validators, err := getValidatorsList(balances)
 	if err != nil {
 		return common.NewErrorf("generate_challenge",
 			"error getting the validators list: %v", err)
 	}
 
+
+	// Check if the length of the list of validators is higher than the lower bound of validators
+	minValidators  := conf.ValidatorsPerChallenge
+	currentValidatorsCount, err := validators.Size(balances);
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/challenge.go]: Please remove the semicolon

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**